contracts/zero-ex/contracts/src/features/multiplex/MultiplexFeature.sol

```diff
@@ -0,0 +1,725 @@
+                _nestedBatchSell(
+                    state,
+                    params,
+                    subcall.data
+                );
+            } else {
+                revert("MultiplexFeature::_executeMultiHopSell/INVALID_SUBCALL");
+            }
+            // The recipient of the current hop will be the source
+            // of tokens for the next hop.
+            state.from = state.to;
+        }
+    }
+
+    function _nestedMultiHopSell(
+        IMultiplexFeature.BatchSellState memory state,
+        IMultiplexFeature.BatchSellParams memory params,
+        bytes memory data,
+        uint256 sellAmount
+    )
+        private
+    {
+        MultiHopSellParams memory multiHopParams;
+        // Decode the tokens and calls for the nested
+        // multi-hop sell.
+        (
+            multiHopParams.tokens,
+            multiHopParams.calls
+        ) = abi.decode(
+            data,
+            (address[], MultiHopSellSubcall[])
+        );
+        multiHopParams.sellAmount = sellAmount;
+        // If the batch sell is using input tokens held by
+        // `address(this)`, then so should the nested
+        // multi-hop sell.
+        multiHopParams.useSelfBalance = params.useSelfBalance;
+        // Likewise, the recipient of the multi-hop sell is
+        // equal to the recipient of its containing batch sell.
+        multiHopParams.recipient = params.recipient;
+        // Execute the nested multi-hop sell.
+        uint256 outputTokenAmount =
+            _executeMultiHopSell(multiHopParams).outputTokenAmount;
+        // Increment the sold and bought amounts.
+        state.soldAmount = state.soldAmount.safeAdd(sellAmount);
+        state.boughtAmount = state.boughtAmount.safeAdd(outputTokenAmount);
+    }
+
+    function _nestedBatchSell(
+        IMultiplexFeature.MultiHopSellState memory state,
+        IMultiplexFeature.MultiHopSellParams memory params,
+        bytes memory data
+    )
+        private
+    {
+        BatchSellParams memory batchSellParams;
+        // Decode the calls for the nested batch sell.
+        batchSellParams.calls = abi.decode(
+            data,
+            (BatchSellSubcall[])
+        );
+        // The input and output tokens of the batch
+        // sell are the current and next tokens in
+        // `params.tokens`, respectively.
+        batchSellParams.inputToken = IERC20TokenV06(
+            params.tokens[state.hopIndex]
+        );
+        batchSellParams.outputToken = IERC20TokenV06(
+            params.tokens[state.hopIndex + 1]
+        );
+        // The `sellAmount` for the batch sell is the
+        // `outputTokenAmount` from the previous hop.
+        batchSellParams.sellAmount = state.outputTokenAmount;
+        // `_computeHopTarget` guarantees that the input
+        // tokens are currently held by `address(this)`,
+        // so `useSelfBalance` is true.
+        batchSellParams.useSelfBalance = true;
```

Comment:
Does it? Even in a `multiplexMultiHopSellTokenForToken([BatchSell, ...])`?

Reply:
good catch 🙈